src/integrations/sessions.rs

```diff
@@ -47,3 +51,16 @@ pub async fn remove_expired_sessions_background_task(
         remove_expired_sessions(gcx.clone()).await;
     }
 }
+
+pub async fn stop_sessions(gcx: Arc<ARwLock<GlobalContext>>) {
+    let mut gcx_locked = gcx.write().await;
+    let keys = gcx_locked.integration_sessions.iter()
+        .map(|(key, _)| key.to_string())
+        .collect::<Vec<_>>();
+
+    for key in keys {
+        if let Some(session) = gcx_locked.integration_sessions.remove(&key) {
+            Box::into_pin(session.lock().await.try_stop()).await;
```

Comment:
It may be safer to add all these removed sesions to a list, and remove them after the global context lock is released, just in case global context may be needed at some point to kill some session.

Some sessions like docker require it to be dropped, in the docker case because it needs to kill containers and access some data that is in global context, this could cause deadlock.